### ISSUE TITLE
[FEATURE_REQUEST] API to quickly chat with Characters from other sites

### ISSUE THREAD

**Issue by cditschler** [SillyTavern/SillyTavern#2473]
### Have you searched for similar requests?

Yes

### Describe the solution you'd like

It would be cool if we could extend any site (e.g. chub.ai) via User-Script or WebExtension such that they have an "open in ST" Button. That button would then connect to a user-defined ST-URL, and use some api (e.g. `/api/add-character`). It could also switch to that character (in ST) and perhaps do other things like loading expression packs etc - all done via one click.

Is there already an API that could be used for this? If not, where would be a good place to add this API?

### Priority

Low (Nice-to-have)

### Are you willing to test this on staging/unstable branch if this is implemented?

None

**Comment by Wolfsblvt** [comment]
Yes, ST has API endpoints to import characters and expression packs.
ST does not have an open URL with arguments though. And I am not sure how that could easily be achievable, as ST only really works in one window. You shouldn't open multiple tabs.

**Comment by cditschler** [comment]
> ST does not have an open URL with arguments though

Ok, but could we make it such that it opens the chat in the existing tab?

**Comment by cditschler** [comment]
> Ok, but could we make it such that it opens the chat in the existing tab?

Or perhaps the last opened one, in case there are multiple. If there isn't such an API, do you think it would be hard to add this? I guess the Frontend would have to open some "contionus channel" over which it could receive commands form the backend, e.g. an EventSource or even WebSocket.

> Yes, ST has API endpoints to import characters and expression packs.

For the record: For relevant routes, see [server.js](https://github.com/SillyTavern/SillyTavern/blob/a410c63333069100aee6d4e4b54196cd6accfa24/server.js) and esarch for `/import`.
I wonder which API it would be for expression packs, though?
